src/main/java/seedu/meal360/storage/Database.java

```diff
@@ -0,0 +1,189 @@
+package seedu.meal360.storage;
+
+import java.io.IOException;
+import java.util.HashMap;
+import seedu.meal360.IngredientList;
+import seedu.meal360.Recipe;
+import seedu.meal360.RecipeList;
+import seedu.meal360.WeeklyPlan;
+
+public class Database {
+
+    private static final String recipesDatabaseFilepath = "." + System.getProperty("file.separator") + "database"
+            + System.getProperty("file.separator")
+            + "recipesDatabase.json";
+
+    private static final String weeklyPlanDatabaseFilepath = "." + System.getProperty("file.separator") + "database"
+            + System.getProperty("file.separator")
+            + "weeklyPlanDatabase.json";
+
+    private static final String userIngredientsDatabaseFilepath = "." + System.getProperty("file.separator")
+            + "database" + System.getProperty(
+                    "file.separator")
+            + "userIngredientsDatabase.json";
+
+    JsonDatabaseHelper<RecipeList> recipeListJsonDatabaseHelper = new JsonDatabaseHelper<>(
+            recipesDatabaseFilepath, defaultRecipeList(), RecipeList.class);
+
+    JsonDatabaseHelper<WeeklyPlan> weeklyPlanJsonDatabaseHelper = new JsonDatabaseHelper<>(
+            weeklyPlanDatabaseFilepath, new WeeklyPlan(), WeeklyPlan.class);
+
+    JsonDatabaseHelper<IngredientList> userIngredientsJsonDatabaseHelper = new JsonDatabaseHelper<>(
+            userIngredientsDatabaseFilepath, new IngredientList(), IngredientList.class);
+
+    public RecipeList loadRecipesDatabase() throws IOException {
+        return recipeListJsonDatabaseHelper.loadDatabase();
+    }
+
+    public void saveRecipesDatabase(RecipeList recipeList) throws IOException {
+        recipeListJsonDatabaseHelper.saveDatabase(recipeList);
+    }
+
+    public WeeklyPlan loadWeeklyPlanDatabase() throws IOException {
+        return weeklyPlanJsonDatabaseHelper.loadDatabase();
+    }
+
+    public void saveWeeklyPlanDatabase(WeeklyPlan weeklyPlan) throws IOException {
+        weeklyPlanJsonDatabaseHelper.saveDatabase(weeklyPlan);
+    }
+
+    public IngredientList loadUserIngredientsDatabase() throws IOException {
+        return userIngredientsJsonDatabaseHelper.loadDatabase();
+    }
+
+    public void saveIngredientsDatabase(IngredientList userIngredients) throws IOException {
+        userIngredientsJsonDatabaseHelper.saveDatabase(userIngredients);
+    }
+
+    public RecipeList defaultRecipeList() {
+        RecipeList defaultRecipeList = new RecipeList();
+        defaultRecipeList.add(new Recipe("Chicken Rice", new HashMap<>() {
+            {
+                put("Chicken", 1);
+                put("Rice", 1);
+                put("Salt", 1);
+                put("Pepper", 1);
+            }
```

Comment:
Perhaps consider refactoring such hashmaps as constants to help reduce the LOC of this method and also give greater clarity as to its purpose: eg i am adding new recipe called CHICKEN_RICE with CHICKEN_RICE_INGREDIENTS (the hashmap)